expression/integration_test.go

```diff
@@ -10208,3 +10208,9 @@ func (s *testIntegrationSuite) TestConstPropNullFunctions(c *C) {
 	tk.MustExec("insert into t2 values (0, 'c', null), (1, null, 0.1), (3, 'b', 0.01), (2, 'q', 0.12), (null, 'a', -0.1), (null, null, null)")
 	tk.MustQuery("select * from t2 where t2.i2=((select count(1) from t1 where t1.i1=t2.i2))").Check(testkit.Rows("1 <nil> 0.1"))
 }
+
+func (s *testIntegrationSuite) TestIssue26977(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	result := tk.MustQuery("select a + 1 as f from (select cast(0xfffffffffffffff0 as unsigned) as a union select cast(1 as unsigned)) t having f != 2;")
+	result.Check(testkit.Rows("18446744073709551601"))
```

Comment:
Could you also add the original test case?

Reply:
Do you mean add this to test case ?
```
tidb> create table t1 (a bigint unsigned);
Query OK, 0 rows affected (0.01 sec)

tidb> insert into t1 values (0xfffffffffffffff0);
Query OK, 1 row affected (0.00 sec)

tidb> insert into t1 select a + 1 from t1;
```

I don’t think this is necessary. The test case I added can show that there was a bug before, and this bug has been fixed in this pr